server/server.go

```diff
@@ -584,7 +596,7 @@ func (s *Server) createClient(conn net.Conn) *client {
 	}
 	// If there is a max connections specified, check that adding
 	// this new client would not push us over the max
-	if s.opts.MaxConn > 0 && len(s.clients) >= s.opts.MaxConn {
+	if s.getOpts().MaxConn > 0 && len(s.clients) >= s.getOpts().MaxConn {
```

Comment:
Do you think it is worth optimizing in this function to snapshot MaxConn, TLS options, and AuthTimeout?  I realize establishing the connection and TLS are already expensive, but am thinking of a thundering herd of clients reconnecting.